src/main/kotlin/org/rust/lang/core/psi/ext/RsQualifiedNamedElement.kt

```diff
@@ -216,22 +216,34 @@ data class RsQualifiedName private constructor(
             return Item(parts[1], type)
         }
 
-        private fun childItem(raw: String): Item? {
+        private fun childItems(raw: String): List<Item>? {
+
+            fun toChildItem(type: String, name: String): Item? {
+                val itemType = ChildItemType.fromString(type) ?: return null
+                return Item(name, itemType)
+            }
+
             val parts = raw.split(".")
-            if (parts.size != 2) return null
-            val type = ChildItemType.fromString(parts[0]) ?: return null
-            return Item(parts[1], type)
+            if (parts.size != 2 && parts.size != 4) return null
+            val result = mutableListOf<Item>()
+
+            result += toChildItem(parts[0], parts[1]) ?: return null
+            if (parts.size == 4) {
+                result += toChildItem(parts[2], parts[3]) ?: return null
+            }
+
+            return result
```

Comment:
```suggestion
            return parts.windowed(size = 2, step = 2).map {
                toChildItem(it.first(), it.last()) ?: return null
            }
```

Reply:
I thought about it too and decided that such approach more complicated for reader since `window` is used quite rare. But I don't have strong opinion in this case. So, if you think it's better, let's rewrite it